api/provider/aws/task_logs.go

```diff
@@ -0,0 +1,17 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (t *TaskProvider) Logs(taskID string, tail int, start, end time.Time) ([]models.LogFile, error) {
```

Comment:
I understanding splitting the tough log logic into a separate file with `logging.go` so it can be re-used by multiple entities, but unless I'm missing something major here, I don't see tests for any of that logic. Seems unwise given how much string manipulation we're doing in that code, etc.

Reply:
these are wrappers for aws api calls, which should be tested in system tests

Reply:
In some cases, the helpers have a decent amount of logic surrounding the API calls. The "pagination" logic in `getLogEvents` could be split out, for example.  And `generateTaskIDMap` is an potentially testable function.

Reply:
Right now, all of the existing api helper functions w/ pagination do so within the helper function itself, and they aren't being unit tested. This would break the RoT and make it really inconsistent what is unit tested (and, in my opinion, be poor design).  So far, we've been testing helpers that only touch layer0 concepts, e.g. tags. 

However, if we do decide certain api calls should be tested, we need to do it on a consistent basis. I don't think this PR is in that scope. 

In regards to `generateTaskIDMap`, I just don't think that unit test would be very useful
from the perspective of "unit tests should give you confidence in your code". It's such a small function, I don't think it needs one.

Reply:
I agree that adding more tests here would break the pattern we have in place, and is therefore not in the scope of this PR. I made a separate issue to discuss that https://github.com/quintilesims/layer0/issues/374